pulsar/src/main/java/io/micronaut/pulsar/intercept/PulsarClientIntroductionAdvice.java

```diff
@@ -116,9 +116,7 @@ public Object intercept(MethodInvocationContext<Object, Object> context) {
 
                 throw new IllegalArgumentException("Pulsar producers can only return MessageId or body being sent.");
             } catch (PulsarClientException e) {
-                if (LOG.isErrorEnabled()) {
-                    LOG.error("Failed to produce message on producer " + producerId, e);
-                }
+                LOG.error("Failed to produce message on producer {}", producerId, e);
```

Comment:
does this print exception properly ? i had issues exception being treated as obj and doing .toString on it instead of printing the exception but not sure which version of loggers

Reply:
it should (and it's easy to test). the library looks at the varargs and if the last one is an exception type and the count of the rest matches the count of `{}` placeholders, those get replaced and the exception gets logged.